art-decisions/proposals/2022-10-26-standardize-python-formatting.md

```diff
@@ -1,6 +1,36 @@
 # Title
 
-We should standardize on python formatting expectations. We should set one standard and adhere to it judiciously without exception. I recommend that we simply adopt the standards already established by the pbench organization and eliminate further debate on the topic. This formatting standard should also be applied wherever possible with automation and CI.
+Standardize on python code formatting requirements
+
+All Python code must conform to [PEP 8](https://peps.python.org/pep-0008), and
+will be checked via the use of [`flake8`](https://github.com/PyCQA/flake8).
+
+All Python imports will be required to be sorted and grouped according to [PEP
+8](https://peps.python.org/pep-0008/#imports) using
+[`isort`](https://github.com/PyCQA/isort).
+
+All Python code (Python 3 assumed) will be required to be auto-formatted via
+[`black`](https://github.com/psf/black).
+
+A pre-commit hook will be offered for use to help verify the above
+requirements before they go through the CI job.
+
+The CI environment will include a series of checks to ensure the above
+requirements are met using the following commands:
+
+```
+$ black --check .
+$ isort --profile=black --check .
+$ flake8 .
+```
```

Comment:
As Matt pointed out, the ordering of the commands is likely to be significant, so that should be standardized as well.

Reply:
There is no ordering for the checks, but there is for how you apply changes.  I have posted an update.